### PR TITLE
Feature/keepassxc patches

### PR DESCRIPTION
**Pull request by phoerious**
I applied our patches we made for KeePassXC to this branch. This includes:

- migration to CMake
- restructuring of the folder structure (I kept the qhttp/qhttp headers folder, though to be a little more compatible)
- inclusion of the http-parser library
- host header fix
- C++11 fix
- compiler warning fix

I excluded the TLS patch, since the dev branch seems to have an TLS server implementation. A client implementation appears to be missing, but I think it would be wiser to re-implement that based on the work done for the server component than to simply apply our own patch.

I was thinking about adding the http-parser library as a submodule, but that would complicate things a lot more when exporting our source code release using git archive (git archive does not support submodules).

### CODE REVIEW

File: http-parser/.gitignore [hunk 1]
```diff
@@ -0,0 +1,30 @@
+/out/
```
Comment (droidmonkey) [inline http-parser/.gitignore]: It this file needed?

File: http-parser/.travis.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+language: c
```
Comment (droidmonkey) [inline http-parser/.travis.yml]: Same, is this needed?

File: http-parser/.gitignore (hunk 1 above)
Comment (phoerious) [inline http-parser/.gitignore]: I don't think so, but it's also no big deal if it's in there.

File: http-parser/.travis.yml (hunk 2 above)
Comment (phoerious) [inline http-parser/.travis.yml]: Probably not. I think we can delete the full Travis file, since we don't need it.

**Pull request merged**